lib/AgentManager.js

```diff
@@ -208,6 +208,7 @@ class AgentManager {
             provisioningOK = true
         } catch (err) {
             warn(`Problem encountered during provisioning: ${err.toString()}`)
+            warn(`Reason: ${err.response?.body}`)
```

Comment:
if `err.response` is empty, this will just print "Reason:" - is that desired?

Maybe better to conditionally print the reason?:

```suggestion
            if (err.response?.body) {
                warn(`Reason: ${err.response.body}`)
            }
```

Happy either way if that was the intention.

Reply:
It should never be empty, and reporting the empty state if it exists is helpful

Reply:
"Should" is the operatave word ;) 

The point is if it is, the `?.` will permit the `warn` to print "Reason:".  Similarly, if `body` is empty, it will print "Reason:"

Perhaps better would be:

```suggestion
            warn(`Reason: ${err.response?.body || 'unknown'}`)
```

or similar to avoid the edge case?